test/rules/src/https_everywhere_checker/http_client.py

```diff
@@ -379,13 +377,11 @@ def fetchHtml(self, url):
 			if httpCode == 0:
 				raise HTTPFetcherError("Pycurl fetch failed for '%s'" % newUrl)
 			elif httpCode in (301, 302, 303, 307):
-				# Parse out the headers and extract location, case-insensitively.
-				# If there are multiple location headers, pick the last one.
-				headers = dict()
-				for k, v in self._headerRe.findall(headerStr):
-					headers[k.lower()] = v
-				location = headers.get('location')
-				if not location:
+				location = None
+				for piece in headerStr.split('\n'):
+					if piece.startswith('Location:'):
```

Comment:
You're right, thanks. Also [RFC 7230](https://tools.ietf.org/html/rfc7230#section-3.2) says header field names are case-insensitive. This should be easy to fix.